### ISSUE TITLE
API docs for compiler code

### ISSUE THREAD

**Issue by crazy4pi314** [microsoft/qsharp-compiler#476]
**Is your feature request related to a problem? Please describe.**
It would be really nice to have easy to find/browse online docs for the Q# internals, especially the compiler.

**Describe the solution you'd like**
All of the API doc comments scraped and added to the API reference on the main docs page here: https://docs.microsoft.com/en-us/quantum/

**Describe alternatives you've considered**
Another docs page/site would also be fine, but harder to find.

**Comment by samarsha** [comment]
I agree that this would be good to have! @bettinaheim mentioned that we are working on getting this set up, but we don't have a timeline yet.
